### PR TITLE
Added location search/Combined location and tag search

### PR DESCRIPTION
**Pull request by deefault**
`media_by_tag` is combined with media that was got by location and called `media_found`
Added function `get_media_by_location_id` which takes a location_id as arg.
location_id_list is passed to InstagramBot() class
source is one location or one tag:
so `max_like_for_one_tag ` changed to `max_like_for_one_source`

If tag_list is empty and location_lid_list is not - it looks only for tags.
If only location_id_list is present it looks for location.
If both, it randomly chooses where to search from and then fills `media_found` list, which previously was `media_by_tag` and repeats the process

### CODE REVIEW

File: example.py [hunk 1]
```diff
@@ -56,8 +57,10 @@
 
     # DON'T USE MODE 5 FOR A LONG PERIOD. YOU RISK YOUR ACCOUNT FROM GETTING BANNED
     ## USE MODE 5 IN BURST MODE, USE IT TO UNFOLLOW PEOPLE AS MANY AS YOU WANT IN SHORT TIME PERIOD
-
-    mode = 0
+    if len(sys.argv) != 2:
```
Comment (deefault) [inline example.py]: now you can pass the mode to console, default is 0

**Comment by luisramirez-m** [comment]
I tested this on an account, and everything works fine in mode 0

**Comment by deefault** [comment]
The only problem is to get location ID: 
You need to manually search a location and then copy location ID.
for example:
https://www.instagram.com/explore/locations/213385402/london-united-kingdom/

**Comment by luisramirez-m** [comment]
Yes, that's what I did. Maybe we have to add it in README.md

**Comment by luisramirez-m** [comment]
In another account I have this problem, I use pythonanywhere.com.

`Except on get_media_by_location(): No JSON object could be decoded`

Note: I test this account with the same code in Mac OSX Terminal and it works. Maybe needs some module?

**Comment by hajjboy95** [comment]
Any Updates on this one ? I think its a great feature to have :D

**Comment by luisramirez-m** [comment]
@hajjboy95 I use this versión on one account, but I haven't comments. Only Likes and Follow/Unfollow.

**Comment by hajjboy95** [comment]
@luisramirez-m How can I target people in Dubai for example using this , only liking/following/unfollowing?

If so what do I need to change from the example.py file?

If you can link me ur current setup for your country your targeting that'll be helpful 👍

**Comment by luisramirez-m** [comment]
You need to download this pullrequest, and on `example.py` you need change this number with Dubai ID Location`location_id_list=['215031298'],`.

If you visit this URL https://www.instagram.com/explore/locations/213281320/dubai-united-arab-emirates/ you see in the URL this number 213281320 and this is Dubai ID Location, you need to put in `location_id_list=['213281320'],` and that is it.

Something like this:
https://gist.github.com/luisramirez-m/6e5b169b785d5bbb6bdab76759a31861

**Comment by hajjboy95** [comment]
@luisramirez-m  ur amazing! , thanks 👍

**Comment by luisramirez-m** [comment]
Thanks to @deefault he created this Pull requests.

**Comment by ChipWolf** [comment]
Add some README commits and I'll review @deefault

**Comment by xb3sox** [comment]
It's the best feature that i was waiting for thanks @deefault 

I tried to run it but it gives me this error 
`Traceback (most recent call last):
  File "C:\Users\user\Downloads\instabot.py\example.py", line 44, in <module>
    unfollow_whitelist=['example_user_1', 'example_user_2'])
TypeError: __init__() got an unexpected keyword argument 'location_id_list'`

any solution ?

**Comment by xb3sox** [comment]
@luisramirez-m @ChipWolf 

any solution ?

**Comment by luisramirez-m** [comment]
@xb3sox Remove example_user_1 and example_user_2 `unfollow_whitelist=['example_user_1', 'example_user_2']` and try again.

**Comment by xb3sox** [comment]
@luisramirez-m it didn't work! :(

**Comment by luisramirez-m** [comment]
You donwload this Pull Request?. You need download this 2 files (https://github.com/instabot-py/instabot.py/pull/508/files) and Replace for the actually (example.py and src/instabot.py) I have 2 bots run this Pull Request since 27 April, and no have problemas, but this Pull Request have problemas with comments settings, comments don't work.

**Comment by xb3sox** [comment]
@luisramirez-m Thanks man its seems i downloaded the old commit on Apr 17, 2017. I downloaded the latest one and it works. Thanks @luisramirez-m and Thanks to @deefault

**Comment by deefault** [comment]
I've done some changes to readme and resolved conflicts

**Comment by Bigemm** [comment]
Is there a version of this already added to a example.py & instabot.py src file?

**Comment by Bigemm** [comment]
Is there a way of telling that the bot is collecting media restricted to that tag like your dubai example? @luisramirez-m

**Comment by mwargan** [comment]
Great pull! Very helpful - just one question... The console prints "Getting media by location ID" but not "Getting media by tags", even though both are set. A quick look at what the bot is doing shows me it's not filtering by the tag list, nor by the tag blacklist. Have I set something wrong up? Also - what's mode 6?

**Comment by remyx11** [comment]
@luisramirez-m sorry im a total noob at this GitHub thing, but i managed to install python, the bot and so on. But i just don't know how to "download" these files to make location thingy work.. Can you help?

**Comment by luisramirez-m** [comment]
@remyx11 I Download files manually https://github.com/instabot-py/instabot.py/pull/508/files if you clic on button named "view" you can view all code:

example.py
https://github.com/deefault/instabot.py/blob/da9b6739f99bdff2abfe2c68d78881284b63c1f1/example.py

instabot.py
https://github.com/deefault/instabot.py/blob/da9b6739f99bdff2abfe2c68d78881284b63c1f1/src/instabot.py

Copy and past on your files named example.py and instabot.py

**Comment by sociallion** [comment]
how do i get it to only like a location with certain hashtags. Right now it will like 'pools' and it will like fort Lauderdale, but not only "pools' in fort Lauderdale. I have customers that only want to like hashtags in their area of business.

**Comment by marfert** [comment]
There is a bug in the logic in line 595 of instabot.py: if tag_list is empty but location_id_list is not (or vice versa), the bot exits with "No tags and location IDs were given".

**Comment by Mishadow98** [comment]
I have problems

What do I do?

Mikhails-MacBook-Pro:instabot.py-remastered mikhailsmirnov$ python example.py 
Traceback (most recent call last):
  File "example.py", line 12, in <module>
    from instabot import InstaBot
  File "/Users/mikhailsmirnov/Desktop/instabot.py-remastered/src/instabot.py", line 363
    def get_media_id_by_location(self, location_id):
                                                   ^
IndentationError: unindent does not match any outer indentation level
Mikhails-MacBook-Pro:instabot.py-remastered mikhailsmirnov$

**Comment by Tidle999** [comment]
so is there any possibilty to combine the location id with hashtags?

**Comment by andrewsegas** [comment]
Guys, here it's working, location and tags together
https://github.com/andrewsegas/instaboost

**Comment by ChipWolf** [comment]
If this is still an issue, please open a new issue. Thanks!

**Pull request closed without merging**